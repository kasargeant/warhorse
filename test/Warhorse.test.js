"use strict";

// Imports
const Warhorse = require.requireActual("../src/core/Warhorse");

let warhorse = new Warhorse({
    "bundle": {
        "minify": true
    }
});
console.log("@@@@@@@" + warhorse.settings.directory);
// beforeAll(function() {
//
// });
//
// describe("Test check.", function() {
//     it("should be able to fail.", function() {
//         expect(false).toBe(true);
//     });
// });


describe("The Warhorse class", function() {

    // SYNCHRONOUS TESTS
    it("should be instantiatable", function() {
        expect(warhorse).toBeDefined();
    });

    it("should be able to parse a filepath", function() {
        let path = warhorse.splitPath("./test/shared/client_src/index.js");
        expect(path.name).toBe("index.js"); // Sanity test - don't rely just on snapshots!!!
        expect(path).toMatchSnapshot();
    });

    // ASYNCHRONOUS TESTS
    it("should be able to load a file", function(done) {


        warhorse.load("./test/shared/client_src/index.js", function(file) {
            if(!file) {done.fail();}
            // console.log(file.name);
            // warhorse.bundle(file, function(file) {
            //     // let dstPath = "../test/client_dist/" + file.name;
            //     // warhorse.save(file, dstPath);
            //     expect(tree).toMatchSnapshot();
            // });
            //console.log(JSON.stringify(file));
            expect(file.content).toMatchSnapshot();

            //expect(file.name).toBe("index.js");
            done();
        });

    });
});

