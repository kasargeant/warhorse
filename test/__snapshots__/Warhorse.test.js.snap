// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Warhorse class should be able to load a file 1`] = `
"// Imports
const Square = require(\\"./Square\\");

let sq = new Square(30);

console.log(sq.area);
"
`;

exports[`The Warhorse class should be able to parse a filepath 1`] = `
Object {
  "config": false,
  "directory": "./test/shared/client_src",
  "extension": ".js",
  "name": "index.js",
  "path": "./test/shared/client_src/index.js",
  "stem": "./test/shared/client_src/index",
}
`;
